dataprep/eda/configs.py

```diff
@@ -0,0 +1,1040 @@
+    """
+    enable: bool, default True
+        Whether to create this element
+    duplicates__threshold: int, default 1
+        The threshold for duplicated row counts
+    similar_distribution__threshold:int, default 0.05
+        The significance level for Kolmogorov–Smirnov test
+    uniform__threshold: int, default 0.999
+        The p-value threshold for chi-square test
+    missing__threshold: int, default 1
+        The threshold for missing values count
+    skewed__threshold: int, default 1e-5
+        The threshold for skewness statistics
+    infinity__threshold: int, default 1
+        The threshold for infinity count
+    zeros__threshold: int, default 5
+        The threshold for zeros count
+    negatives__threshold: int, default 1
+        The threshold for negatives count
+    normal__threshold: int, default 0.99
+        The p-value threshold for normaltest, it is based on D’Agostino and Pearson’s test that
+        combines skew and kurtosis to produce an omnibus test of normality
+    high_cardinality__threshold: int, default 50
+        The threshold for unique values count, count larger than threshold yields high cardinality
+    constant__threshold: int, default 1
+        The threshold for unique values count, count equals to threshold yields constant value
+    outstanding_no1__threshold: int, default 1.5
+        The threshold for outstanding no1 insight, measures the ratio of the largest category count
+        to the second-largest category count
+    attribution__threshold: int, default 0.5
+        The threshold for the attribution insight, measures the percentage of the top 2 categories
+    high_word_cardinality__threshold: int, default 1000
+        The threshold for the high word cardinality insight, which measures the number of words of
+        that cateogory
+    outstanding_no1_word__threshold: int, default 0
+        The threshold for the outstanding no1 word threshold, which measures the ratio of the most
+        frequent word count to the second most frequent word count
+    outlier__threshold: int, default 0
+        The threshold for the outlier count in the box plot, default 0
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    def __init__(self) -> None:
+        self.enable = True
+        self.duplicates__threshold = 1
+        self.similar_distribution__threshold = 0.05
+        self.uniform__threshold = 0.999
+        self.missing__threshold = 1
+        self.skewed__threshold = 1e-5
+        self.infinity__threshold = 1
+        self.zeros__threshold = 5
+        self.negatives__threshold = 1
+        self.normal__threshold = 0.99
+        self.high_cardinality__threshold = 50
+        self.constant__threshold = 1
+        self.outstanding_no1__threshold = 1.5
+        self.attribution__threshold = 0.5
+        self.high_word_cardinality__threshold = 1000
+        self.outstanding_no1_word__threshold = 1.5
+        self.outlier__threshold = 0
+
+
+class Hist:
```

Comment:
I would suggest put each type of class into a separate file. E.g. under folder `dataprep/eda/vis_types`. This would make it easier to add new visualization types. Besides, in the future we may also want to put all related functions into corresponding vis type file, which may make code more manageable.